packages/evm/src/eof/container.ts

```diff
@@ -246,6 +246,7 @@ class EOFHeader {
     // Write the start of the first code section into `codeStartPos`
     // Note: in EVM, if one would set the Program Counter to this byte, it would start executing the bytecode of the first code section
     this.codeStartPos = [relativeOffset]
+    this.getCodePosition(this.codeSizes.length - 1) // initialize code positions
```

Comment:
Ah no, we should not do this. The cache itself is not initialized because this is an optimization. This will set the code sizes and the code positions for all code sections. So if we have a call to an EOF container we would (with this change) initialize all these relative offsets even if only the first code section is ran. So if you do the inverse (get code section from a `pc`) then this is done in "debug mode" - it is fine if we are slower there. But for "normal" mode we want to take out any calculations which we don't need (like this). We would only want to find the code position when we really need it. (Let me know if this is still unclear, can push a fix here)

Reply:
I'll address this in an EOF optimization PR later.